Fix local devnet python script

Doy-lee opened the pull request:
There were multiple road-blocking bugs preventing the devnet python script from working which I've worked through

- I updated the contract ABIs hardcoded in the ethereum script (would be nice to automate this)
- Added separate devnet parameters
- Update the parsing code for the service node blob received from the smart contract
- Fix an incorrect negation on `vector::empty()` causing an invalid iterator to be dereferenced and crash a non-service node daemon.

There are still failing unit-tests that are to be investigated. Some of them are because some codepaths require an ethereum provider to be set mandatorily even if they're not a service node.

Doy-lee committed:
Set ETH devnet parameters seperately

Doy-lee committed:
Correctly parse out new Service Node BLOB from contract

Doy-lee committed:
Update contract ABI on ethereum.py script

Doy-lee committed:
Decrease sleep time before getting bls rewards from 155 to 10s

Doy-lee committed:
Fix incorrect not on empty() check for RPC clients

Doy-lee committed:
Simplify error check in rewards contract, move string helper to utils

Doy-lee committed:
Parse out contributors in the SN contract blobs

darcys22 commented:
Yeah nice! 

I suspect this will cause the devnet that jason has spun up on the integration branch to crash however. Sort of the nature of both being devnet but having separate params needed.

solution might just be commenting out the local devnet params but leave them in there, and when we are working locally we can uncomment them out

Doy-lee committed:
Add USE_LOCAL_DEVNET_ETH_ADDRESSES option to opt into devnet config

Doy-lee commented:
Came back around to this, I was gonna add a `private-devnet` flag which forced me to look at the command line stuff which is super complicated and abstracted. So I backed that out, but I had some simplifications to that part of the codebase so I went there and did that and deleted a bunch of code.

I've added the cmake option `USE_LOCAL_DEVNET_ETH_ADDRESSES` option to CMake which enables the addresses that I hardcoded in this PR so it runs with the local-devnet python script.  By default that's off, so it should allow our _live_ devnet to continue operating.

Doy-lee committed:
Simplify command line, N=1 is same as N=many, use C++20 requires to merge struct sfinae

Doy-lee commented:
Backed out the `std::optional` change for the default values in command line because that's a bit viral and touches everything and some code relies on the subtle behaviour it uses that's pretty error prone to try and fix.

This gets the unit tests running again and failing at consensus error which is the thing I've been investigating/fixing.
